pkg/registry/core/service/portallocator/controller/repair.go

```diff
@@ -112,14 +112,13 @@ func (c *Repair) doRunOnce() error {
 	// If etcd server is not running we should wait for some time and fail only then. This is particularly
 	// important when we start apiserver and etcd at the same time.
 	var snapshot *api.RangeAllocation
-
-	err := wait.PollImmediate(time.Second, 10*time.Second, func() (bool, error) {
-		var err error
-		snapshot, err = c.alloc.Get()
-		return err == nil, err
+	var lastError error
+	err := wait.PollUntilContextTimeout(context.Background(), 2*time.Second, 30*time.Second, true, func(ctx context.Context) (done bool, err error) {
+		snapshot, lastError = c.alloc.Get()
+		return err == nil, nil
 	})
 	if err != nil {
-		return fmt.Errorf("unable to refresh the port allocations: %v", err)
```

Comment:
If the log is printed here, when the timeout in other places causes Fatal(such as etcd timeout), this place may have not yet returned, resulting in no log output.

Reply:
If Fatal occur, this is not main cause of Fatal, it is not important to print log here.

Reply:
Emm, how about this approach?
```go
	if err != nil {
                 if lastError != nil {
			return fmt.Errorf("unable to refresh the port allocations: %w", lastError)
		}
		return fmt.Errorf("unable to refresh the port allocations: %w", err)
	}
```

Reply:
People maybe want to know there are something failed in backstage, silence would make problem hard to locate.

Reply:
Where the Fatal occuer will print fatal log. The code before has not fixed this, also the code where used wait.PollUntilContextTimeout do not fix this more. Print log every time in loop retry?